## Calculating Covariance and Correlation

Consider a portfolio composed of *Walmart* and *Facebook*. Do you expect the returns of these companies to show high or low covariance? Or, could you guess what the correlation would be? Will it be closer to 0 or closer to 1? 

Begin by extracting data for Walmart and Facebook from the 1st of January 2014 until today.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
mydata = pd.read_csv('Dataset\Walmart_FB_2014_2017.csv', index_col = 'Date')
mydata

,WMT,FB
Date,,
2013-12-31,71.126411,54.650002
2014-01-02,71.325287,54.709999
2014-01-03,71.090279,54.560001
2014-01-06,70.692558,57.200001
2014-01-07,70.909485,57.919998
...,...,...
2017-04-03,70.906502,142.279999
2017-04-04,71.084190,141.729996
2017-04-05,70.728821,141.850006


Repeat the process we went through in the lecture for these two stocks. How would you explain the difference between their means and their standard deviations?

In [4]:
sec_returns = np.log(mydata / mydata.shift(1))
print (sec_returns.head())

                 WMT        FB
Date                          
2013-12-31       NaN       NaN
2014-01-02  0.002792  0.001097
2014-01-03 -0.003300 -0.002745
2014-01-06 -0.005610  0.047253
2014-01-07  0.003064  0.012509


In [7]:
sec_returns[['WMT', 'FB']].mean() * 250

WMT    0.003551
FB     0.287439
dtype: float64

In [8]:
sec_returns[['WMT', 'FB']].std() * 250 ** 0.5

WMT    0.177607
FB     0.287726
dtype: float64

***

## Covariance and Correlation


\begin{eqnarray*}
Covariance Matrix: \  \   
\Sigma = \begin{bmatrix}
        \sigma_{1}^2 \ \sigma_{12} \ \dots \ \sigma_{1I} \\
        \sigma_{21} \ \sigma_{2}^2 \ \dots \ \sigma_{2I} \\
        \vdots \ \vdots \ \ddots \ \vdots \\
        \sigma_{I1} \ \sigma_{I2} \ \dots \ \sigma_{I}^2
    \end{bmatrix}
\end{eqnarray*}

Covariance matrix:

In [11]:
cov_matrix = sec_returns.cov() * 250
cov_matrix

,WMT,FB
WMT,0.031544,0.008777
FB,0.008777,0.082786


__Interpretacion de resultados__

Covarianza de WMT: 0.031544
Covarianza de FB: 0.082786

Covarianza entre WMT y FB: 0.008777

Correlation matrix:

In [10]:
corr_matrix = sec_returns.corr()
corr_matrix

,WMT,FB
WMT,1.000000,0.171746
FB,0.171746,1.000000


__Interpretacion de resultados__

Correlacion de retornos entre WMT y FB: 0.171746

Would you consider investing in such a portfolio?